### ISSUE TITLE
Allowing open a single file directly

### ISSUE THREAD

**Issue by wzy1935** [lapce/lapce#796]
I want to open a single file directly without open a directory first. For example, I hope running `lapce ./README.md` can directly open this file like other editors do, but so far it gives no response in my computer(Windows 11). I'm not sure is it a bug or unimplemented feature?

**Comment by panekj** [comment]
lapce cannot open single files currently

**Comment by graphik55** [comment]
+1 this missing feature stops me to use Lapce as my daily editor.
I open a lot of single file over a day from different place and right now it is to uncomfortable :'(

**Comment by schwrzstrbn** [comment]
hoping to see this feature soon<3

**Comment by mrjbq7** [comment]
I'm trying to add Lapce support as an editor for the Factor programming language, and we need this feature added for that to work.

It would also be nice to be able to open a single file to a particular line number, some other editors have a convention to specify line number like (e.g., ``lapce file.txt:10``) when opening.

**Comment by panekj** [comment]
> I'm trying to add Lapce support as an editor for the Factor programming language, and we need this feature added for that to work.

Posting about needing the feature won't make it happen faster.

**Comment by mrjbq7** [comment]
> Posting about needing the feature won't make it happen faster.

Well, yes, but I added the context of being able to open a file to a line number, and I've also started working on a branch to add the feature myself after a brief discussion on Discord.

**Comment by panekj** [comment]
> Well, yes, but I added the context of being able to open a file to a line number

This was more of a general message to everyone, I guess I should have quoted everyone.


> I've also started working on a branch to add the feature myself after a brief discussion on Discord.

What's the approach? Because I already discussed it with Dzhou and I'm working on it.

**Comment by mrjbq7** [comment]
> What's the approach? Because I already discussed it with Dzhou and I'm working on it.

I'm happy to let you build it -- I'm trying to learn the codebase, but was going to add code into ``LapceData::load`` that would try and find an appropriate window to open the file into if you already had a workspace appropriate for the path, and if not, create an empty workspace, then open the file into that.

**Comment by mrjbq7** [comment]
This sort-of works, for example, in ``LapceData::load``:

```rust
        if let Some(path) = path {
            #[cfg(target_os = "windows")]
            let workspace_type =
                if !env::var("WSL_DISTRO_NAME").unwrap_or_default().is_empty()
                    || !env::var("WSL_INTEROP").unwrap_or_default().is_empty()
                {
                    LapceWorkspaceType::RemoteWSL
                } else {
                    LapceWorkspaceType::Local
                };

            #[cfg(not(target_os = "windows"))]
            let workspace_type = LapceWorkspaceType::Local;

            let path = PathBuf::from(path).canonicalize().unwrap();
            if path.is_dir() {
                let info = WindowInfo {
                    size: Size::new(800.0, 600.0),
                    pos: Point::new(0.0, 0.0),
                    maximised: false,
                    tabs: TabsInfo {
                        active_tab: 0,
                        workspaces: vec![LapceWorkspace {
                            kind: workspace_type,
                            path: Some(path),
                            last_open: 0,
                        }],
                    },
                };

                let window = LapceWindowData::new(
                    keypress.clone(),
                    panel_orders.clone(),
                    event_sink.clone(),
                    &info,
                    db.clone(),
                );
                windows.insert(window.window_id, window);

            } else {

                let info = WindowInfo {
                    size: Size::new(800.0, 600.0),
                    pos: Point::new(0.0, 0.0),
                    maximised: false,
                    tabs: TabsInfo {
                        active_tab: 0,
                        workspaces: vec![LapceWorkspace {
                            kind: workspace_type,
                            path: None,
                            last_open: 0,
                        }],
                    },
                };

                let window = LapceWindowData::new(
                    keypress.clone(),
                    panel_orders.clone(),
                    event_sink.clone(),
                    &info,
                    db.clone(),
                );
                windows.insert(window.window_id, window);

                println!("opening file");
                let _ = event_sink.submit_command(
                    LAPCE_UI_COMMAND,
                    LapceUICommand::OpenFile(path),
                    Target::Auto
                );
            }
```

**Comment by panekj** [comment]
> The open at a line number required a bit more thought, I didn't see how to send that as a command with the available code.

afaik, it's not supported by lapce yet, would be better to open separate feature request for it since it would be for cli/palette

**Comment by panekj** [comment]
fixed in #1003
